infra/elasticsearch/src/utils/index.ts

```diff
@@ -0,0 +1,188 @@
+//
+
+import type { Client } from "@elastic/elasticsearch";
+import { logger as cdtnLoggger } from "@socialgouv/cdtn...infra...logger";
+
+import type { Document } from "../types";
+import { analyzer, char_filter, filter, tokenizer } from "./analysis";
+
+//
+
+export type Mappings = Record<string, string>;
+export interface UtilsOptions {
+  client: Client;
+  indexName: string;
+}
+
+//
+
+export const logger = cdtnLoggger.child({
+  package: "@socialgouv/cdtn...infra...elasticsearch",
+});
+
+//
+
+export async function createIndex({
+  client,
+  indexName,
+  mappings,
+}: UtilsOptions & { mappings: Mappings }): Promise<void> {
+  const { body } = await client.indices.exists({ index: indexName });
+  if (body) {
+    try {
+      await client.indices.delete({ index: indexName });
+      logger.info(`Index ${indexName} deleted.`);
+    } catch (error: unknown) {
+      logger.error("index delete", error);
```

Comment:
I think we can raise the error here. If we can't delete the existing indices, why should we continue?

Reply:
Perhaps improve the log with an action: `Failed to  delete index...`